extensions/wikia/AdEngine/js/provider/gpt/helper.js

```diff
@@ -22,53 +22,29 @@ define('ext.wikia.adEngine.provider.gpt.helper', [
 	adDetect,
 	AdElement,
 	GoogleTag,
+	recoveryHelper,
 	slotTweaker,
-	SourcePointTag,
 	sraHelper,
 	scrollHandler
 ) {
 	'use strict';
 
 	var logGroup = 'ext.wikia.adEngine.provider.gpt.helper',
-		context = adContext.getContext(),
 		googleApi = new GoogleTag(),
-		slotsToRecover = [],
 		sourcePointInitialized = false;
 
-	function isRecoveryEnabled() {
-		return !!(context.opts.sourcePoint && SourcePointTag);
-	}
-
-	function isBlocking() {
-		return !!(isRecoveryEnabled() && window.ads && window.ads.runtime.sp.blocking);
-	}
-
-	function isRecoverable(slotName, recoverableSlots) {
-		return isRecoveryEnabled() && recoverableSlots.indexOf(slotName) !== -1;
-	}
-
-	function recoverSlots() {
-		if (!isBlocking()) {
-			return;
-		}
-		log(['Starting recovery', slotsToRecover], 'debug', logGroup);
-		while (slotsToRecover.length){
-			window.ads.runtime.sp.slots.push([slotsToRecover.shift()]);
-		}
-	}
-
 	function loadRecovery() {
 		if (sourcePointInitialized) {
 			return;
 		}
 		log('SourcePoint recovery enabled', 'debug', logGroup);
 		sourcePointInitialized = true;
-		googleApi = new SourcePointTag();
-		recoverSlots();
+		googleApi = recoveryHelper.createSourcePointTag();
+		recoveryHelper.recoverSlots();
 	}
 
 	function loadSourcePoint() {
-		if (isBlocking()) {
+		if (recoveryHelper.isBlocking() && recoveryHelper.isRecoveryEnabled()) {
```

Comment:
Maybe it's worth to add:
```
if (!recoveryHelper.isRecoveryEnabled()) {
  return;
}
```
above this if statement - to prevent adding listener on `sp.blocking` when recovery is disabled or we can remove `isRecoveryEnabled()` since loadSourcePoint is executed only then (https://github.com/Wikia/app/pull/8533/files#diff-d0456a56fa8978685b777a3c10aaf5abR129).